### ISSUE TITLE
Decode a query param again on config change

### ISSUE THREAD

**Issue by kirilldronkin** [pbeshai/use-query-params#50]
This will be very useful for cases when a config created dynamically, something like this:

```js
const PageParam = useMemo(() => RangeParam.create(1, pagesCount), [pagesCount]);

const [page = 1, setPage] = useQueryParam('page', PageParam);
```

Here `RangeParam` is a custom `QueryParamConfig` which decodes a value only when it falls into the given range, otherwise `undefined` is returned and the default page (`1`) is used. Looks like config should be passed to the `useMemo` and `useCallback` dependencies in `useQueryParam` hook to achieve this behaviour.

**Comment by pbeshai** [comment]
I think I happened to fix this while addressing #46. Can you test in 0.4.2 and see if it is resolved?

**Comment by kirilldronkin** [comment]
Yes, it was resolved in 0.4.2, thank you.
